### PR TITLE
Update Docker 1.13.0, machine 0.9.0

### PR DESCRIPTION
**Pull request by StefanScherer**
Draft a release 1.2.0:

* Update to Docker 1.13.0
* Update to Docker Machine 0.9.0

Docker Compose still on 1.9.0 is compilation seems difficult (again).

### CODE REVIEW

File: builder/test-integration/spec/hypriotos-image/docker_spec.rb [hunk 1]
```diff
@@ -86,13 +86,13 @@
...
 describe command('docker -v') do
-  its(:stdout) { should match /Docker version 1.12.5, build/ }
+  its(:stdout) { should match /Docker version 1.13.0, build/ }
   its(:exit_status) { should eq 0 }
 end
 
 describe command('docker version') do
-  its(:stdout) { should match /Client:. Version:      1.12.5. API version:  1.24/m }
-  its(:stdout) { should match /Server:. Version:      1.12.5. API version:  1.24/m }
+  its(:stdout) { should match /Client:. Version:      1.13.0. API version:  1.24/m }
```
Comment (DieterReuter) [inline builder/test-integration/spec/hypriotos-image/docker_spec.rb]: neu API version is 1.25 not 1.24
```
API version:  1.25 (minimum version 1.12)
```

**Comment by DieterReuter** [comment]
@StefanScherer can you please add another fix? Docker 1.13.0 default to `overlay2` storage driver.'
```
@@ -65,7 +65,7 @@ describe file('/var/lib/docker') do
   it { should be_owned_by 'root' }
 end

-describe file('/var/lib/docker/overlay') do
+describe file('/var/lib/docker/overlay2') do
   it { should be_directory }
   it { should be_mode 700 }
   it { should be_owned_by 'root' }
```

**Comment by StefanScherer** [comment]
Should https://github.com/hypriot/image-builder-rpi/blob/cae34451393e6780f647d1d4747ac1fc9aacf708/builder/test-integration/spec/hypriotos-image/docker_spec.rb#L53 also be changed to overlay2?

**Comment by StefanScherer** [comment]
Ah, handled in #154 by removing it :-)

File: builder/test-integration/spec/hypriotos-image/filesystems_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
   it { should be_loaded }
 end
 
-Specinfra::Runner.run_command('modprobe overlay')
-describe kernel_module('overlay') do
+Specinfra::Runner.run_command('modprobe overlay2')
```
Comment (StefanScherer) [inline builder/test-integration/spec/hypriotos-image/filesystems_spec.rb]: @DieterReuter I've also changed this test. If Docker 1.13.0 uses overlay2 we should check for overlay2, right?

Comment (DieterReuter) [inline builder/test-integration/spec/hypriotos-image/filesystems_spec.rb]: No, kernel module is just 'overlay'.

**Comment by DieterReuter** [comment]
@StefanScherer please revert/remove last commit "Check overlay2", then it's fine to merge.

**Comment by StefanScherer** [comment]
@DieterReuter done, let's see what Travis does...

**Comment by DieterReuter** [comment]
LGTM

**Comment by MathiasRenner** [comment]
Dear sender,

I changed my email address (mail@mathias-renner.de) to:

mail@mathias-renner.com

Please use the new address in the future.

I received your recent email. You don't need to re-send it.

Best,
Mathias Renner

**Pull request merged**